blog/2024-07-19-coordinator_async_setup.md

```diff
@@ -0,0 +1,77 @@
+---
+author: Josef Zweck
+authorURL: https://github.com/zweckj
+title: Initialize a `DataUpdateCoordinator` with `_async_setup`
+---
+
+In Home Assistant 2024.8, we are introducing the `_async_setup` method for coordinators. 
+This method allows you to run asynchronous code to prepare your `DataUpdateCoordinator`
+or to load data that only needs to be loaded once.
+
+You can override `_async_setup` in your coordinator, and it will be automatically
+called during `coordinator.async_config_entry_first_refresh()`.
+It offers the same error handling as `_async_update_data` and will handle `ConfigEntryError`
+and `ConfigEntryAuthFailed` accordingly.
+
+## Example
+
+```python
+# Example of a custom DataUpdateCoordinator with the `_async_setup` method
+# This example demonstrates how to initialize data that only needs to be loaded once.
+
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+class MyUpdateCoordinator(DataUpdateCoordinator[MyDataType]):
+    """Class to handle fetching data from the API centrally."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+    ) -> None:
+        """Initialize coordinator."""
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+        self.prereq_data: SomeData | None = None
+        self.my_api = MyApi()
+
+
+    async def _async_setup(self) -> None:
+        """Do initialization logic."""
+        self.prereq_data = await self.my_api.get_initial_data()
+
+    async def _async_update(self) -> MyDataType:
+        """Do the usual update"""
+        return await self.my_api.update(self.prereq_data)
```

Comment:
Note that `self.prereq_data` is still optional here so we may still need a condition check or hint the type checker that `self.prereq_data` isn't `None` anymore.

Reply:
moved it to class attribute, now mypy would be happy